### ISSUE TITLE
`JaggedTensor.from_data_indices_and_list_ids` ldim=2 issue

### ISSUE THREAD

**Issue by swahtz** [openvdb/fvdb-core#356]
I believe that when we create a `JaggedTensor` of ldim=2 from `from_data_indices_and_list_ids`, that there's an issue with how the outer dim length is constructed.  Here is an example:

```
import torch
import fvdb


ch_dim = 1
# Create a JaggedTensor of ldm2 out of a list of [3, 2, 4] tensors
jt = fvdb.JaggedTensor([[torch.rand([3,ch_dim]),torch.empty([0,ch_dim]),torch.rand([15,ch_dim])],
                        [torch.rand([10,ch_dim]),torch.empty([8,ch_dim])],
                        [torch.rand([5,ch_dim]),torch.empty([4,ch_dim]),torch.rand([10,ch_dim]),torch.rand([1,ch_dim])]])
print(jtsum.jlidx)

jt2 = fvdb.JaggedTensor.from_data_indices_and_list_ids(jt.jdata, jt.jidx, jt.jlidx, torch.max(jt.jidx).item()+1)

print(len(jt)) # 3
print(len(jt2)) # 9

```

From my understanding, jt2 should have been constructed to be the same as jt but the lenght of jt will be 3 (the correct outer dimension of the length of list of tensors) and jt2's length will be 9 (the length of the `num_tensors` argument), which is incorrect.
